packages/core-components/src/components/Avatar/Avatar.tsx

```diff
@@ -86,19 +89,21 @@ export function Avatar(props: AvatarProps) {
       ...customStyles,
     };
   }
+  const fontStyles = {
+    fontFamily: styles.fontFamily,
+    fontSize: styles.fontSize,
+    fontWeight: styles.fontWeight,
+  };
```

Comment:
Hmm, wonder if this might have a performance impact by causing re-generation of styles on every render or not?

Reply:
Yeah, it's unfortunate, I tried wrapping in a `useMemo` but we can't do that because `makeStyles` is already a hook, and there's no way to override the diffing. The ID's aren't based off the contents of the `style` properties which is unfortunate, I guess because they would be too big and cause a lot of page weight.

I wonder if we should just break the API so that you pass in `classNames` instead, and use `makeStyles` yourself in the parent component?